markdown/docs/user-guide.md

```diff
@@ -28,370 +37,455 @@ You can start a FastBoot server on your development machine by running:
-seamlessly in both environments.
+  // ... Application code
+});
+```
 
-[ember-network]: https://github.com/tomdale/ember-network
-[fetch-api]: https://developer.mozilla.org/en-US/docs/Web/API/Fetch_API
+This service contains several useful objects and methods you can use to manage an application's state when it is being FastBooted.
 
-To use `ember-network`, install it as you would any addon:
+Property         | Description
+---------------- | ---------------------------------------------------------------------------------------------------------
+`isFastBoot`     | Equals `true` if your application is running in FastBoot
+`response`       | The FastBoot server's response
+`request`        | The request sent to the FastBoot server
+`shoebox`        | A key/value store for passing data acquired server-side to the client
+`deferRendering` | A function that takes a `Promise` that you can use to defer the Ember Applications rendering in FastBoot
 
-```sh
-ember install ember-network
-```
+### Deferring Response Rendering
 
-Once installed, you can import it using the JavaScript module syntax,
-wherever you need to make a network request:
+By default, FastBoot waits for the `beforeModel`, `model`, and `afterModel` hooks to resolve before sending a response back to the client.
+You can use these hooks to defer rendering of your application.
+See [Defer Rendering with Model Hooks](#defer-rendering-with-model-hooks).
 
-```js
-import fetch from 'ember-network/fetch';
-```
+If you have asynchronous code that runs outside of these lifecycle hooks, you will want to use `deferRendering` to block the response. `deferRendering` function accepts a `Promise` and will chain all promises passed to it.
+FastBoot will wait for these promises to resolve before sending the response to the client.
 
-The `fetch()` method returns a promise and is very similar to jQuery's
-`getJSON()` method that you are likely already using. For example,
-here's an Ember route that uses `fetch()` to access the GitHub JSON API
-and use it as the route's model:
+You must call `deferRendering` before these model hooks complete.
+For example, if you made an asynchronous call in a Component, you would use `deferRendering` in the `init` lifecycle hook.
 
-```js
-import Route from 'ember-route';
-import fetch from 'ember-network/fetch';
+### FastBoot Request
+
+The `fastboot.request` key allows you access to the request sent to the FastBoot server.
+
+#### Access Request Headers
+
+You can access the current request headers via `fastboot.request`.
+The `headers` object implements part of the [Fetch API's Headers class](https://developer.mozilla.org/en-US/docs/Web/API/Headers), the functions available are [`has`](https://developer.mozilla.org/en-US/docs/Web/API/Headers/has), [`get`](https://developer.mozilla.org/en-US/docs/Web/API/Headers/get), and [`getAll`](https://developer.mozilla.org/en-US/docs/Web/API/Headers/getAll).
```

Comment:
We can, but if we are going to start explaining headers I think it would e better to include a direct link to the Mozilla docs: https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers